xmodule/library_tools.py

```diff
@@ -173,63 +135,28 @@ def update_children(self, dest_block, user_perms=None, version=None):
-        exact same definition ID used in the copy block.
-
-        This method will update dest_block's 'source_library_version' field to
-        store the version number of the libraries used, so we easily determine
-        if dest_block is up to date or not.
         """
         if user_perms and not user_perms.can_write(dest_block.location.course_key):
             raise PermissionDenied()
-
-        if not dest_block.source_library_id:
-            dest_block.source_library_version = ""
-            return
-
-        source_blocks = []
-
         library_key = dest_block.source_library_key
         is_v2_lib = isinstance(library_key, LibraryLocatorV2)
-
         if version and not is_v2_lib:
             library_key = library_key.replace(branch=ModuleStoreEnum.BranchName.library, version_guid=version)
 
         library = self._get_library(library_key, is_v2_lib)
         if library is None:
             raise ValueError(f"Requested library {library_key} not found.")
 
-        if hasattr(dest_block, 'capa_type'):
-            filter_children = (dest_block.capa_type != ANY_CAPA_TYPE_VALUE)
-        else:
-            filter_children = None
-
         if not is_v2_lib:
             if user_perms and not user_perms.can_read(library_key):
                 raise PermissionDenied()
-            if filter_children:
-                # Apply simple filtering based on CAPA problem types:
-                source_blocks.extend(self._problem_type_filter(library, dest_block.capa_type))
-            else:
-                source_blocks.extend(library.children)
+        update_children_task.delay(self.user_id, str(dest_block.location), version)
 
-            with self.store.bulk_operations(dest_block.location.course_key):
-                dest_block.source_library_version = str(library.location.library_key.version_guid)
-                self.store.update_item(dest_block, self.user_id)
-                head_validation = not version
-                dest_block.children = self.store.copy_from_template(
-                    source_blocks, dest_block.location, self.user_id, head_validation=head_validation
-                )
-                # ^-- copy_from_template updates the children in the DB
-                # but we must also set .children here to avoid overwriting the DB again
-        else:
-            # TODO: add filtering by capa_type when V2 library will support different problem types
-            source_blocks = library_api.get_library_blocks(library_key, block_types=None)
-            source_block_ids = [str(block.usage_key) for block in source_blocks]
-            dest_block.source_library_version = str(library.version)
-            self.store.update_item(dest_block, self.user_id)
-            self.import_from_blockstore(dest_block, source_block_ids)
+    def import_task_status(self, dest_block):
+        """
+        Return task status for update_children_task.
+        """
+        return get_import_task_status(dest_block.location)
```

Comment:
Is it good idea to change parameter name to location? And when function is used it will be 
`self.tools. import_task_status(self.location)` to not pass all block